### DISCUSSION TITLE
receiving and transmitting a file using BPSK

### DISCUSSION THREAD

**Discussion by JohnCopler** [duggabe/gr-control#16] [Q&A] [answered]
Good afternoon We are trying to transfer a text file using BSK modulation. 
We transmit the signal at a frequency 5.8Ghz,  sampling frequency 8e4 and usrp_rate/hackrf_rate 8e5. Can you tell me what the problem might be?
![hasil receiver](https://github.com/user-attachments/assets/9e3acca0-d89f-4e1d-8bf4-fe3090330ce1)
![Receiver](https://github.com/user-attachments/assets/7c1adc5d-f55f-4a65-a67b-668f6c4f621e)
![tx](https://github.com/user-attachments/assets/acba5aed-ff8e-4911-a4fa-9d5a85f80371)

**Comment by duggabe** [comment, marked as answer]
You need to check the specs for the HackRF. I think it needs a sample rate of 8MHz (8e6).
